cst438_register/src/main/java/com/cst438/controller/StudentController.java

```diff
@@ -40,20 +42,20 @@ public Student addStudent(@RequestParam("email")String email, @RequestParam("nam
 	}
 	
 	@PostMapping("/student/hold")
-	public Student updateHold(@RequestParam("email")String email, @RequestParam("addhold")boolean addhold){
-		Student s = studentRepository.findByEmail(email);
-		if(s!=null) {//student exists in database
+	public Optional<Student> updateHold(@RequestParam("id")int id, @RequestParam("addhold")boolean addhold){
+		Optional<Student> s = studentRepository.findById(id);
+		if(s.isPresent()) {//student exists in database
 			if(addhold) {
-				s.setStatusCode(1);
-				s.setStatus("Hold");
-				studentRepository.save(s);
+				s.get().setStatusCode(1);
```

Comment:
I would consider creating a student variable assigned to s.get() to help reduce the amount of function calls and make code a little cleaner

Reply:
I'm not sure how to go about this because student is optional in this case, will look into it!